docs/SalesforceStreaming-streamingsource.md

```diff
@@ -82,8 +82,8 @@ The Salesforce types will be automatically mapped to schema types as shown below
 | long        | _long                                                                      |              |
 | double      | _double, currency, percent                                                 |              |
 | date        | date                                                                       |              |
-| timestamp   | datetime                                                                   | Milliseconds |
-| time        | time                                                                       | Milliseconds |
+| timestamp   | datetime                                                                   | Microseconds |
```

Comment:
I have added a couple of tests and updated existing tests to ensure they check correct date time types usage:
`SalesforceSchemaUtilTest.testGetSchemaWithFields`
`SalesforceSchemaUtilTest.testValidateSupportedFieldSchemas` 
`SalesforceSchemaUtilTest.testCheckCompatibilityIncorrectLogicalType`
`SalesforceRecordReaderTest.testTypes`

Reply:
But in all these tests you have manually set it to be timestamp micros, which means the same mistake could have been made. Is there a way to check that time is micros or millis coming from salesforce?

Reply:
Timestamp from Salesforce comes in String format, so we just parse using `Instant.parse` method, then convert it to mircoseconds.

Reply:
I see, so it's not specified by salesforce. Why do we switch it from millis to micros here?

Reply:
I have switched to using mircos instead of millis for two reasons:
1. CDAP UI does not distinguish between two types of timestamp (millis and mircos) and only supports timestamp micros. When user defines timestamp type on UI, it is send to plugin as timestamp-micros. If we set timestamp-millis in schema JSON and import to UI, such timestamp is detected as long and logical type is lost.
2. Micros is more exact than millis. Thus we ensure we won't loose anything during conversion.

Reply:
Ok sg